library/Pdfexport/HeadlessChrome.php

```diff
@@ -247,106 +264,169 @@ public function fromHtml($html, $asFile = false)
-                            $this->remote[0],
-                            $this->remote[1],
-                            $e
-                        );
-                    }
+        $pdf = '';
+        // We don't intend to register any then/otherwise handlers, so call done on that promise
+        // to properly propagate unhandled exceptions to the caller.
+        $this->toPdfCommon()->done(
+            function (string $newPdf) use (&$pdf) {
+                $pdf = $newPdf;
+            },
+            function (Throwable $e) {
+                if (! $e instanceof HeadlessChromeException) {
+                    throw $e;
                 }
 
-                // Fallback to the local binary if a remote chrome is unavailable
-            case $this->binary !== null:
-                $browserHome = $this->getFileStorage()->resolvePath('HOME');
-                $commandLine = join(' ', [
-                    escapeshellarg($this->getBinary()),
-                    static::renderArgumentList([
-                        '--bwsi',
-                        '--headless',
-                        '--disable-gpu',
-                        '--no-sandbox',
-                        '--no-first-run',
-                        '--disable-dev-shm-usage',
-                        '--remote-debugging-port=0',
-                        '--homedir=' => $browserHome,
-                        '--user-data-dir=' => $browserHome
-                    ])
-                ]);
-
-                if (Platform::isLinux()) {
-                    Logger::debug('Starting browser process: HOME=%s exec %s', $browserHome, $commandLine);
-                    $chrome = new Process('exec ' . $commandLine, null, ['HOME' => $browserHome]);
-                } else {
-                    Logger::debug('Starting browser process: %s', $commandLine);
-                    $chrome = new Process($commandLine);
-                }
+                Logger::error($e);
+                Logger::debug($e->getTraceAsString());
+            }
+        );
 
-                $loop = Factory::create();
+        Loop::run();
 
-                $killer = $loop->addTimer(10, function (TimerInterface $timer) use ($chrome) {
-                    $chrome->terminate(6); // SIGABRT
-                    Logger::error(
-                        'Terminated browser process after %d seconds elapsed without the expected output',
-                        $timer->getInterval()
-                    );
-                });
+        return $pdf;
+    }
 
-                $chrome->start($loop);
+    protected function toPdfCommon(): ExtendedPromiseInterface
+    {
+        $deferred = new Promise\Deferred();
+        Loop::futureTick(function () use ($deferred) {
+            switch (true) {
+                case $this->remote !== null:
+                    try {
+                        $result = $this->jsonVersion($this->remote[0], $this->remote[1]);
+                        if (is_array($result)) {
```

Comment:
If this isn't true, you have to handle this case and reject the promise. Remember to do this outside the try-block.